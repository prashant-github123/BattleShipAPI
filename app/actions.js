import request from 'axios';
import {
  ADD_PLAYER,
  SHIP_COORDS,
  SHIP_LOCATION,
  STATUS,
    ADD_NEW_USER_URI,
    PLACE_SHIP_URI,
    RETRIVE_SHIP_LOC_URI,
    CHECK_TURN_URI
 } from './constants';
import {browserHistory} from 'react-router';

export const addPlayerAPI = (playerName) => { // eslint-disable-line
  console.log(playerName);
  const data = {
    playerName: playerName,
  }
  return request({
    method: 'POST',
    url: ADD_NEW_USER_URI,
    data,
  });
};

export const retrieveShipLocationsAPI = (gameId) => { // eslint-disable-line
  const data = {
    gameId : gameId,
  }
  return request({
    method: 'POST',
    url: RETRIVE_SHIP_LOC_URI,
    data,
  });
};

export const placeShipAPI = (playerData, cords) => { // eslint-disable-line
  console.log(playerData);
  const data = {
    shipCoordinates: cords.toString(),
    gameId: playerData.gameID,
    playerId: playerData.playerID,
    playerName: playerData.playerName,
  };
  return request({
    method: 'POST',
    url: PLACE_SHIP_URI,
    data,
  });
};

export const checkTurnStatusAPI = (gamedata) => {
  console.log(gamedata);
  return request({
    method: 'POST',
    url: CHECK_TURN_URI,
    data: gamedata,
  });
};

export const setCurrentUser = (data) => {
  return {
    type: ADD_PLAYER,
    data,
  };
};

export const setShipCoordinates = (data) => { // eslint-disable-line
  return {
    type: SHIP_COORDS,
    data,
  };
};

export const setShipLocation = (data) => { // eslint-disable-line
  return {
    type: SHIP_LOCATION,
    data,
  };
};

export const setTurnStatus = (data) => { // eslint-disable-line
  return {
    type: STATUS,
    data,
  };
};

/*
* Function addPlayer
* @param : playerName
*/
export const addPlayer = (playerName) => { //eslint-disable-line
  return function (dispatch) {
    addPlayerAPI(playerName).then((resp) => { // eslint-disable-line
      console.log(resp);
      dispatch(setCurrentUser(resp.data));
      window.location.href = '/placeShip';
    }).catch((err) => {
      Error(err);
    });
  };
};
/*
* Function placeShip
* @param : playerData : Player Information
* @param : Cords : Ship Co-ordinates
*/
export const placeShip = (playerData, cords) => { // eslint-disable-line
  console.log(playerData, cords);
  return function () {
    placeShipAPI(playerData, cords).then((resp) => { // eslint-disable-line
      window.location.href = '/gameBoard';
    }).catch((err) => {
      Error(err);
    });
  };
};
/*
* Function retriveShipLocation
* @param : gameId
*/
export const retriveShipLocation = (gameId) => {// eslint-disable-line
  return function (dispatch) {
    retrieveShipLocationsAPI(gameId).then((response) => {
      dispatch(setShipLocation(response.data));
      return response.data;
    }).catch((err) => {
      console.log("Error", err);
    });
  };
};
/*
* Function checkTurnStatus
* @param : data
*/
export const checkMyTurnStatus = (data) => { // eslint-disable-line

  return checkTurnStatusAPI(data).then((response) => {
    return response.data;
  }).catch((err) => {
    console.log("Error", err);
  });
};

/*Start New Game*/
export const startNew = () => {
  sessionStorage.clear();
  browserHistory.push('/');
}
