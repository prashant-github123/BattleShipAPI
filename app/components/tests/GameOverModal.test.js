import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import GameoverModal from './../GameOverModal';

import {startNew} from './../../actions'
// Snapshot for GameoverModal React Component
describe('GameoverModal --- Snapshot',()=>{
    it('capturing Snapshot of GameoverModal', () => {
        const winner = 'test';
        const renderedValue =  renderer.create(<GameoverModal 
                                                    startNew={startNew}
                                                    winner ={winner}/>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('GameoverModal --- Shallow Render REACT COMPONENTS',()=>{
    let wrapper;

    beforeEach(()=>{
        const winner = 'test';
        const startNewMock = jest.fn();
        wrapper = shallow(<GameoverModal  startNew={startNewMock}
                                          winner ={winner}/>)
        
    })

    it('should render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });
    it('should contains button ', () => {
        expect(wrapper.find('button').type()).toEqual('button')
    });
    it('should call startNewGame on click of button',() => {
       wrapper.startNewGame = jest.fn();
        wrapper.find('button').simulate('click');
        wrapper.startNewGame();
        expect(wrapper.startNewGame).toHaveBeenCalled();
    })    
});
