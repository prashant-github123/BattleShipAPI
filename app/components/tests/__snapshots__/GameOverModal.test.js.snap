// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameoverModal --- Snapshot capturing Snapshot of GameoverModal 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "height": "100vh",
      "left": "0",
      "paddingTop": "20%",
      "position": "absolute",
      "textAlign": "center",
      "top": "0",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#fff",
        "fontFamily": "Arial",
        "fontSize": "2.4rem",
        "margin": "0 auto",
        "padding": "5%",
        "width": "50%",
      }
    }
  >
    Game Over!!!
    <br />
    test
    <br />
    <button
      className="next-submit"
      onClick={[Function]}
    >
      Start New Game
    </button>
  </div>
</div>
`;
